cpp/src/reductions/segmented_reductions.cpp

```diff
@@ -32,19 +32,37 @@ struct segmented_reduce_dispatch_functor {
   device_span<size_type const> offsets;
   data_type output_dtype;
   null_policy null_handling;
+  std::optional<const scalar*> init;
   rmm::mr::device_memory_resource* mr;
   rmm::cuda_stream_view stream;
 
   segmented_reduce_dispatch_functor(column_view const& segmented_values,
                                     device_span<size_type const> offsets,
                                     data_type output_dtype,
                                     null_policy null_handling,
+                                    std::optional<const scalar*> init,
                                     rmm::cuda_stream_view stream,
                                     rmm::mr::device_memory_resource* mr)
     : col(segmented_values),
       offsets(offsets),
       output_dtype(output_dtype),
       null_handling(null_handling),
+      init(init),
+      mr(mr),
+      stream(stream)
+  {
+  }
```

Comment:
Separate with a blank line.
```suggestion
  }

```